### PR TITLE
C++ casting changes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/http.cpp [hunk 1]
```diff
@@ -269,22 +269,22 @@ int Exiv2::http(Exiv2::Dictionary& request, Exiv2::Dictionary& response, std::st
...
+    memset(reinterpret_cast<char*>(&serv_addr), 0, serv_len);
 
     serv_addr.sin_addr.s_addr = inet_addr(servername_p);
     serv_addr.sin_family = AF_INET;
     serv_addr.sin_port = htons(atoi(port_p));
 
     // convert unknown servername into IP address
     // http://publib.boulder.ibm.com/infocenter/iseries/v5r3/index.jsp?topic=/rzab6/rzab6uafinet.htm
-    if (serv_addr.sin_addr.s_addr == (unsigned long)INADDR_NONE) {
+    if (serv_addr.sin_addr.s_addr == static_cast<unsigned long> INADDR_NONE) {
```
Comment (piponazo) [inline src/http.cpp:287]: [blocker] There is a typo here. `static_cast<>` is missing the parenthesis around `INADDR_NONE`. Interestingly this was only caught on CI by the OSX builds:

```
/Users/travis/build/Exiv2/exiv2/src/http.cpp:287:64: error: expected '(' after 'static_cast'
    if (serv_addr.sin_addr.s_addr == static_cast<unsigned long> INADDR_NONE) {
```

**Review by piponazo — CHANGES_REQUESTED** [review]
Thanks again! You're on fire 🔥 

The changes look good but I have pointed you out one minor mistake detected by the CI. I'll approve and merge once you fix that little issue.

File: src/http.cpp (hunk 1 above)
Comment (neheb) [reply src/http.cpp:287]: Compiled fine locally. I’ll fix as soon as I can.

Comment (clanmills) [reply src/http.cpp:287]: Wow.  This is quite a fine.  On Monday I was investigating an ASAN is with gethostbyname() on the next line of code.  I haven't decide yet what to do about this yet, however it's getting thought and attention.

I'll deal with ASAN on 0.27-maintenance.  We only started to exercise this code in the test suite with 0.27.3

```
    serv_addr.sin_addr.s_addr   = inet_addr(servername_p);
    serv_addr.sin_family        = AF_INET    ;
    serv_addr.sin_port          = htons(atoi(port_p));

    // convert unknown servername into IP address
    // http://publib.boulder.ibm.com/infocenter/iseries/v5r3/index.jsp?topic=/rzab6/rzab6uafinet.htm
    if (serv_addr.sin_addr.s_addr == (unsigned long)INADDR_NONE)
    {
        struct hostent* host = gethostbyname(servername_p);   // <-- ASAN when executed from python, OK from bash 
        if ( !host )  return error(errors, "no such host", servername_p);
        memcpy(&serv_addr.sin_addr,host->h_addr,sizeof(serv_addr.sin_addr));
    }
```

Comment (neheb) [reply src/http.cpp]: I think I figured out why it was compiling...

There are two )) that are needed at the end. Basically, static_cast<unsigned long> INADDR_NONE were evaluated separately.

**Comment by neheb** [comment]
clang-tidy got rid of the cast here:

```
            auto tl = (unsigned long)((size / 16) * 16);
            auto tl_offset = (unsigned long)(size - tl);
```

I'm curious if this is even correct. unsigned long = 32-bits in Windows (both x86 and x64). It varies on Linux.

**Comment by neheb** [comment]
Bah auto is not supported in function parameters. Would be helpful in cleaning up some Windows vs Linux stuff.

**Review by piponazo — APPROVED** [review]
Now it is looking better 👏 

I'll merge it when all the CI checks finish

File: src/http.cpp (hunk 1 above)
Comment (clanmills) [reply src/http.cpp]: I've logged my RemoteIo/ASAN discoveries: #1307

**Pull request merged**